doc/manual/src/language/constructs.md

```diff
@@ -217,18 +217,20 @@ three kinds of patterns:
     
     > **Warning**
     > 
-    > The `args@` expression is bound to the argument passed to the
-    > function which means that attributes with defaults that aren't
-    > explicitly specified in the function call won't cause an
-    > evaluation error, but won't exist in `args`.
+    > The `args@` expression is bound to the argument that is passed to the
+    > function. This does not take default values into account.
+    >
+    > Attributes with default values that aren't
+    > specified in the function call won't cause an
+    > evaluation error. But they also won't exist in `args`.
     > 
     > For instance
     > 
     > ```nix
     > let
-    >   function = args@{ a ? 23, ... }: args;
+    >   f = args@{ a ? 23, ... }: args;
     > in
-    >   function {}
+    >   f {}
     > ````
```

Comment:
```suggestion
    > ````
    >
    > is equivalent to
    >
    > ````
    >   f = args: [ (args.a or 23) args ];
    > in
    >   f {}
    > ````
```

Reply:
The functions are not equivalent. Destructuring lambdas are spine strict. While you're not say that they are, you'd lead people to believe that they are.

```
nix-repl> lib.fix (args: { b = args.a; a = 1; })  
{ a = 1; b = 1; }

nix-repl> lib.fix ({a, ...}: { b = a; a = 1; })
error:
       … from call site

         at /home/user/h/nixpkgs/lib/fixed-points.nix:19:20:

           18|   # details.
           19|   fix = f: let x = f x; in x;
             |                    ^
           20|

       error: infinite recursion encountered

       at «none»:0: (source not available)
```

Reply:
Oh good point I did not consider that at all!
```suggestion
    > ````
    >
    > is equivalent to:
    >
    > ````nix
    >   f = args @ { ... }: [ (args.a or 23) args ];
    > in
    >   f {}
    > ````
```
This however should be correct?